### PR TITLE
Jit64AsmCommon: Make frsqrte and fres PIE-compliant

### PR DESCRIPTION
**Pull request by MerryMage**
There were two strategies I had for this:

1. Using the stack (`push` / `pop`)
2. Rematerializing the exponent from the value in `xmm0`.

I chose the former strategy because it had the smallest codesize and was the least error-prone. Codesize increase of +6 bytes each.

### CODE REVIEW

File: Source/Core/Core/PowerPC/Jit64Common/ConstantPool.cpp [hunk 1]
```diff
@@ -59,5 +59,5 @@ Gen::OpArg ConstantPool::GetConstantOpArg(const void* value, size_t element_size
   _assert_msg_(DYNA_REC, info.m_size == value_size,
                "Constant has incorrect size in constant pool.");
   u8* location = static_cast<u8*>(info.m_location);
-  return Gen::M(location + element_size * index);
+  return location + element_size * index;
```
Comment (MerryMage) [inline Source/Core/Core/PowerPC/Jit64Common/ConstantPool.cpp]: Removal of `Gen::M` makes `#include "Common/x64Emitter.h"` irrelevant. Also remember to remove the forward declaration in the header file.

File: Source/Core/Common/MathUtil.cpp [hunk 2]
```diff
@@ -154,21 +153,20 @@ double ApproximateReciprocalSquareRoot(double val)
   int i = (int)(mantissa >> 37);
   vali = sign | exponent;
   int index = i / 2048 + (odd_exponent ? 16 : 0);
-  vali |= (s64)(frsqrte_expected_base[index] - frsqrte_expected_dec[index] * (i % 2048)) << 26;
+  auto& entry = frsqrte_expected[index];
```
Comment (lioncash) [inline Source/Core/Common/MathUtil.cpp]: const reference

File: Source/Core/Common/MathUtil.cpp [hunk 3]
```diff
@@ -213,9 +211,9 @@ double ApproximateReciprocal(double val)
   exponent = (0x7FDLL << 52) - exponent;
 
   int i = (int)(mantissa >> 37);
+  auto& entry = fres_expected[i / 1024];
```
Comment (lioncash) [inline Source/Core/Common/MathUtil.cpp]: const reference.

**Comment by degasus** [comment]
assembly LGTM

**Pull request merged**